src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -53323,6 +53323,52 @@ class B { public static implicit operator C(B b) => new C(); }
             comp.VerifyDiagnostics();
         }
 
+        [Fact, WorkItem(58369, "https://github.com/dotnet/roslyn/issues/58369")]
+        public void ConditionalOperator_TargetTyped_11()
+        {
+            var comp = CreateCompilation(@"
+#pragma warning disable CS0164
+class C<T>
+{
+    static void M2(bool b, string? s)
+    {
+WarnHere:   _ = C.GetCT(s) + (b ? new C1() : new C2());
+ButNotHere: _ = (b ? new C3() : new C4()) + C.GetCT(s);
+    }
+    
+    public static C<T> operator + (C<T> x, C<T> y)
+    {
+        throw null!;
+    }
+}
+
+class C
+{
+    public static C<T> GetCT<T>(T x) => new C<T>(); 
+}
+
+class C1 : C<string> {}
+class C2 : C<string> {}
+class C3 : C<string> {}
+class C4 : C<string> {}
+", options: WithNullableEnable(TestOptions.ReleaseDll));
+
+            comp.VerifyDiagnostics(
+                // (7,35): warning CS8619: Nullability of reference types in value of type 'C1' doesn't match target type 'C<string?>'.
+                // WarnHere:   _ = C.GetCT(s) + (b ? new C1() : new C2());
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "new C1()").WithArguments("C1", "C<string?>").WithLocation(7, 35),
+                // (7,46): warning CS8619: Nullability of reference types in value of type 'C2' doesn't match target type 'C<string?>'.
+                // WarnHere:   _ = C.GetCT(s) + (b ? new C1() : new C2());
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "new C2()").WithArguments("C2", "C<string?>").WithLocation(7, 46),
+                // (8,22): warning CS8619: Nullability of reference types in value of type 'C3' doesn't match target type 'C<string?>'.
+                // ButNotHere: _ = (b ? new C3() : new C4()) + C.GetCT(s);
```

Comment:
>ButNotHere [](http://example.com/codeflow?start=19&length=10)

Are these warnings correct?

Reply:
The warnings, match my expectations. The labels are not meant to convey expectations, the base line is. The code matches the repro used in #58369.